packages/cli/src/index.ts

```diff
@@ -786,8 +798,10 @@ const main = async () => {
     return 1;
   }
 
-  // specifically don't await this, we want to fire and forget
-  telemetryEventStore.save();
+  // The telemetry flush event is called by `telemetryEventStore.save`, and it reinvokes the `main` function
+  if (subSubCommand !== 'flush') {
+    await telemetryEventStore.save();
```

Comment:
This seems brittle. How about spawning the subprocess with the disable telemetry env var instead?

Reply:
Agreed that it feels a bit brittle. We do have a `enabled` check inside the `.save` function as things are right now so an env var would prevent us from sending. 

https://github.com/vercel/vercel/pull/12555/files#diff-2d5a21aecc56e19e130f744d9c268cf35ff1de827416e704d27483a01ffbd94fR232